src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -49,6 +49,11 @@ internal abstract partial class AnalyzerDriver : IDisposable
         /// </summary>
         private readonly bool _hasDiagnosticSuppressors;
 
+        /// <summary>
+        /// Filtered diagnostic severities in the compilation, i.e. diagnostics with effective severity from this set should not be reported.
```

Comment:
It's not that they shouldn't be reported though but that they should not be run at all correct?

Reply:
There are 2 effects here:

1. Performance: Do not execute analyzers that only report hidden/info diagnostics.
2. Functional: End users should never see hidden/info diagnostics from the execution.

These overlap for majority of scenarios, except if an analyzer has more then one SupportedDiagnostics, such that one is one warning diagnostic and one is hidden/info diagnostic. We cannot skip analyzer execution due to the warning diagnostic, and still need the post-filtering logic that drops the hidden/info diagnostics reported by this analyzer.

Basically, we can guarantee the functional aspect, and do our best in terms of performance aspect. For majority of analyzers, your statement holds true, but I did not want to add the performance aspect to the comment due to the fact that it cannot always be true.

Reply:
I added an additional "PERF" part to the doc comment.